Source/bmalloc/bmalloc/TZoneHeapManager.cpp

```diff
@@ -509,6 +510,27 @@ BINLINE unsigned TZoneHeapManager::tzoneBucketForKey(const TZoneSpecification& s
     return bucket;
 }
 
+#if BUSE_DYNAMIC_TZONE_COMPACTION
+bool g_tzoneDynamicCompactionEnabled = false;
+#endif
+
+void TZoneHeapManager::initializeTZoneDynamicCompaction()
+{
+#if BUSE_DYNAMIC_TZONE_COMPACTION
+    RELEASE_BASSERT(s_state >= State::Seeded);
+    const char* envEnable = getenv("bmalloc_DynamicallyCompactTZoneAllocations");
+    if (envEnable && (!strcasecmp(envEnable, "true") || !strcasecmp(envEnable, "yes") || !strcmp(envEnable, "1"))) {
+        g_tzoneDynamicCompactionEnabled = true;
+
+        while (!m_tzoneDynamicCompactionSalt)
+            m_tzoneDynamicCompactionSalt = WeakRandom::generate(m_tzoneKeySeed);
```

Comment:
Marcus, this cannot possibly work.  If `WeakRandom::generate(m_tzoneKeySeed)` produces 0, this is going to loop forever.